### PR TITLE
Fix notifications tests

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-13632
affects: student
release note: none

Also adds a way to enable experimental features in UI tests.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Core/Core/UITestHelpers/UITestHelpers.swift [hunk 1]
```diff
@@ -147,6 +152,10 @@ public class UITestHelpers {
             ()
         case .mockNow(let date):
             Clock.mockNow(date)
+        case .enableExperimentalFeatures(let features):
+            for feature in features {
+                ExperimentalFeature(rawValue: feature)!.isEnabled = true
+            }
```
Comment (cobbal) [inline Core/Core/UITestHelpers/UITestHelpers.swift]: There should be a way to reset this so it doesn't mess up neighboring tests. Either in reset() or disabling unset features here would probably work

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1013105&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/UITestHelpers/UITestHelpers.swift [hunk 2]
```diff
@@ -33,9 +33,10 @@ public class UITestHelpers {
         case setAnimationsEnabled(Bool)
         case useMocksOnly
         case debug(Any?)
+        case enableExperimentalFeatures([String])
```
Comment (cobbal) [inline Core/Core/UITestHelpers/UITestHelpers.swift]: Is there a reason not to serialize the `ExperimentalFeature` directly?

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-13632](https://instructure.atlassian.net/browse/MBL-13632)

<details><summary>:x: XCTest failed: CoreTests/GetActivitiesRequestTests/testQuery</summary>


    Assertion Failure: APIActivityTests.swift:36: XCTAssertEqual failed: ("[]") is not equal to ("[Core.APIQueryItem.value("per_page", "10")]")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**